Doc/library/zipfile.rst

```diff
@@ -145,16 +147,25 @@ ZipFile Objects
    adding a ZIP archive to another file (such as :file:`python.exe`).  If
    *mode* is ``'a'`` and the file does not exist at all, it is created.
    If *mode* is ``'r'`` or ``'a'``, the file should be seekable.
+
    *compression* is the ZIP compression method to use when writing the archive,
    and should be :const:`ZIP_STORED`, :const:`ZIP_DEFLATED`,
    :const:`ZIP_BZIP2` or :const:`ZIP_LZMA`; unrecognized
-   values will cause :exc:`NotImplementedError` to be raised.  If :const:`ZIP_DEFLATED`,
-   :const:`ZIP_BZIP2` or :const:`ZIP_LZMA` is specified but the corresponding module
-   (:mod:`zlib`, :mod:`bz2` or :mod:`lzma`) is not available, :exc:`RuntimeError`
-   is raised. The default is :const:`ZIP_STORED`.  If *allowZip64* is
-   ``True`` (the default) zipfile will create ZIP files that use the ZIP64
-   extensions when the zipfile is larger than 4 GiB. If it is  false :mod:`zipfile`
-   will raise an exception when the ZIP file would require ZIP64 extensions.
+   values will cause :exc:`NotImplementedError` to be raised.  If
+   :const:`ZIP_DEFLATED`, :const:`ZIP_BZIP2` or :const:`ZIP_LZMA` is specified
+   but the corresponding module (:mod:`zlib`, :mod:`bz2` or :mod:`lzma`) is not
+   available, :exc:`RuntimeError` is raised. The default is :const:`ZIP_STORED`.
+
+   If *allowZip64* is ``True`` (the default) zipfile will create ZIP files that
+   use the ZIP64 extensions when the zipfile is larger than 4 GiB. If it is
+   ``false`` :mod:`zipfile` will raise an exception when the ZIP file would
+   require ZIP64 extensions.
+
+   The *compresslevel* parameter controls the compression level to use when
+   writing files to the archive. When using :const:`ZIP_STORED` or
+   :const:`ZIP_LZMA` it has no effect. When using :const:`ZIP_DEFLATED`
+   integers ``0`` through ``9`` are accepted. When using :const:`ZIP_BZIP2`
+   integers ``1`` through ``9`` are accepted.
```

Comment:
Maybe it is worth linking this to the _compresslevel_ parameters of the ``:class:`gzip.GzipFile` `` and ``:class:`bz2.Bz2File` `` constructors, which explain the acceptable values.

Reply:
Hmm, I wonder if `zlib.compressobj` and  `bz2.BZ2Compressor` would be more appropriate? I don't want to give the false impression that `GzipFile` or `Bz2File` are actually used at all.

Reply:
I went with your suggestion; the link to the `zlib.compressobj()` function looks weird when rendered.

Reply:
Sorry, yeah _zlib.compressobj_ would have been more appropriate (I was confused; the _gzip_ format is not even used in zip files, just the underlying Deflate algorithm). My point was that you should explain what the different values mean, compared to None. How do the different values control the compression level relative to the default of None?

Reply:
Right, yeah. I think links to the places where this is already described make sense, as you suggest. (we could add that to `tarfile` in another PR also; it's missing the explanation)

Current version looks reasonable when rendered:

![image](https://user-images.githubusercontent.com/1922815/35523287-1ab6f570-04e4-11e8-9c02-5adfcc636778.png)